### PR TITLE
feat(core): service provider - allow custom evm and eth rpc urls

### PR DESCRIPTION
**Pull request by lykalabrada**
#### Which issue(s) does this PR fixes?:

Fixes DFC-343

#### Developer Checklist:

- [ ] Read your code changes at least once
- [ ] Tested on iOS/Android device (e.g, No crashes, library supported etc.)
- [ ] No console errors on web
- [ ] Tested on Light mode and Dark mode\*
- [ ] Your UI implementation visually matched the rendered design\*
- [ ] Unit tests\*
- [ ] Added e2e tests\*
- [ ] Added translations\*

### CODE REVIEW

File: mobile-app/app/contexts/CustomServiceProvider.tsx [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+import React, {
+  createContext,
+  PropsWithChildren,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from "react";
+
+export type CustomServiceProviderType = "dvm" | "evm" | "ethrpc";
```
Comment (fullstackninja864) [inline mobile-app/app/contexts/CustomServiceProvider.tsx:13]: Lets make it enum

File: mobile-app/app/api/wallet/service_provider.ts [hunk 2]
```diff
@@ -1,14 +1,31 @@
...
 
-async function get(): Promise<string | undefined> {
-  const val = await SecuredStoreAPI.getItem(KEY);
-  return val != null ? val : undefined;
+async function get(
+  type: CustomServiceProviderType = CustomServiceProviderType.DVM,
+): Promise<string | undefined> {
+  const PROVIDER_KEY = `${KEY}.${type}`;
+  const val = await SecuredStoreAPI.getItem(PROVIDER_KEY);
+
```
Comment (fullstackninja864) [inline mobile-app/app/api/wallet/service_provider.ts:18]: if we find value here then lets just return it, no need to check one more condition, What you think @lykalabrada ?

File: mobile-app/app/screens/AppNavigator/screens/Settings/components/CustomUrlInput.tsx [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+        testID="endpoint_url_title"
+        text={translate("screens/ServiceProviderScreen", label)}
+      />
+      <View style={tailwind("flex flex-row items-baseline w-full")}>
+        <WalletTextInputV2
+          valid={errMsg === undefined}
+          editable={isUnlocked}
+          value={inputValue.url}
+          inputType="default"
+          onChangeText={(_text: string) => handleUrlInputChange(type, _text)}
```
Comment (fullstackninja864) [inline mobile-app/app/screens/AppNavigator/screens/Settings/components/CustomUrlInput.tsx:87]:
```suggestion
          onChangeText={(text: string) => handleUrlInputChange(type, text)}
```
can we change it to this?

File: mobile-app/app/screens/AppNavigator/screens/Settings/screens/ServiceProviderScreen.tsx [hunk 4]
```diff
@@ -48,152 +108,113 @@ export function ServiceProviderScreen({ navigation }: Props): JSX.Element {
...
-            size={18}
+      <View>
+        {customProviders.map((provider) => (
+          <CustomUrlInput
+            {...provider}
+            inputValue={urlInputValues[provider.type]}
+            activeInput={activeInput}
+            setActiveInput={setActiveInput}
+            setShowActionButtons={setShowActionButtons}
+            handleUrlInputChange={handleUrlInputChange}
```
Comment (fullstackninja864) [inline mobile-app/app/screens/AppNavigator/screens/Settings/screens/ServiceProviderScreen.tsx:179]: key missing?

**Review by fullstackninja864 — CHANGES_REQUESTED** [review]

File: mobile-app/app/api/wallet/service_provider.ts (hunk 2 above)
Comment (lykalabrada) [reply mobile-app/app/api/wallet/service_provider.ts:18]: Agree, that's better 🚀

**Review by fullstackninja864 — APPROVED** [review]

**Review by pierregee — APPROVED** [review]

**Pull request merged**